safe/impact_functions/base.py

```diff
@@ -1408,7 +1408,10 @@ def _calculate_impact(self):
         start_time = datetime.now()
 
         # Run the IF. self.run() is defined in each IF.
-        result_layer = self.run()
+        try:
+            result_layer = self.run()
+        except KeywordNotFoundError as e:
+            raise KeywordNotFoundError(e)
```

Comment:
You are right now ;-)
You don't need to catch and re-raise the exception at every levels.
Something like this will work : 
```
try:
    func1()
except Exception:
    print 'error'


def func1():
    func2()


def func2():
    func3()


def func3():
    raise Exception('test error')
```

To sum up, in InaSAFE, we should just raise the exception where the problem occurs, and then finally catch it in UI to show the problem to the user.